mne/io/fiff/tests/test_raw.py

```diff
@@ -888,9 +888,16 @@ def test_hilbert():
     picks = picks_meg[:4]
 
     raw2 = raw.copy()
+    raw3 = raw.copy()
+    raw4 = raw.copy()
     raw.apply_hilbert(picks)
     raw2.apply_hilbert(picks, envelope=True, n_jobs=2)
 
+    # Test custom N
+    raw3.apply_hilbert(picks, N=raw3.n_times + 100)
+    assert_equal(raw3._data.shape, raw._data.shape)
```

Comment:
if you high pass the data or use a window it shouldn't make a big
difference. When you pad you are basically doing spectrum oversampling yet
using circular convolutions (hence the edge artifacts spreading)